api/routers/notes_router.js

```diff
@@ -0,0 +1,62 @@
+const express = require('express');
+const Notes = require('./../models/notes_model');
+const { validateUserId, restrictUser, checkToken } = require('./../middlewares/user_middleware');
+const { validateNote, validateNoteId } = require('./../middlewares/note_middleware');
+
+const router = express.Router();
+
+router.post('/:userId/notes', validateUserId, validateNote, checkToken, (req, res, next) => {
+
+    const {title, content} = req.body;
+    const userId = req.decoded.id;
```

Comment:
deci id-ul utilizatorului il scoti din tokenul decodat, mai ai nevoie oare in pathname: the acel `:userId`?

Reply:
Eu am nevoie de :userId in middleware-ul validateUserId, ca sa vad daca exista asa utilizator inregistrat in baza.

Reply:
pai okay, de ce sa nu iai acel id mai bine din token-ul decodat? De ce ai avea 2 modalitati diferite de a lua un id